gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -288,6 +267,25 @@ protected void startUp() {
     //Do nothing.
   }
 
+  /**
+   * Method to submit a {@link Dag} to the {@link DagManager} and delete adhoc flowSpecs from the FlowCatalog after
+   * persisting it in the other addDag method called. The DagManager's failure recovery method ensures the flow will be
+   * executed in the event of downtime.
+   * @param flowSpec
+   * @param dag
+   * @param persist
+   * @param setStatus
+   * @throws IOException
+   */
+  public synchronized void addDag(FlowSpec flowSpec, Dag<JobExecutionPlan> dag, boolean persist, boolean setStatus)
```

Comment:
Is this function being called? I see the function below is using a different addDag signature. You probably want to specify a different naming convention here in that case.